src/build_queue.rs

```diff
@@ -726,9 +729,75 @@ pub async fn queue_rebuilds(
     Ok(())
 }
 
+/// Queue rebuilds for failed crates due to a faulty version of rustdoc
+///
+/// It is assumed that the version of rustdoc matches the one of rustc, which is persisted in the DB.
+/// The priority of the resulting rebuild requests will be lower than previously failed builds.
+/// If a crate is already queued to be rebuilt, it will not be requeued.
+/// Start date is inclusive, end date is exclusive.
+#[instrument(skip_all)]
+pub async fn queue_rebuilds_faulty_rustdoc(
+    conn: &mut sqlx::PgConnection,
+    build_queue: &AsyncBuildQueue,
+    start_nightly_date: &NaiveDate,
+    end_nightly_date: &Option<NaiveDate>,
+) -> Result<i32> {
+    let end_nightly_date =
+        end_nightly_date.unwrap_or_else(|| start_nightly_date.succ_opt().unwrap());
+    let mut results = sqlx::query!(
+        r#"
+SELECT c.name,
+       r.version AS "version: Version"
+FROM crates AS c
+         JOIN releases AS r
+              ON c.latest_version_id = r.id
```

Comment:
This join would only queue rebuilds for 

1. releases with docs (which we want)
2. but only the latest release of each crate 

IMO when we rebuild for a broken nightly, we also have to fix older releases, even where this a newer one. 


```suggestion
              ON c.id = r.crate_id
```